File: android/app/src/main/kotlin/com/taajsingh/gurbanilens/ui/home/HomeScreen.kt
package com.taajsingh.gurbanilens.ui.home

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Mic
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.taajsingh.gurbanilens.ui.theme.GurbaniLensTheme

@Composable
fun HomeScreen(
    onSearchTap: () -> Unit,
    onSettingsTap: () -> Unit,
) {
    Scaffold(
        topBar = {
            @OptIn(androidx.compose.material3.ExperimentalMaterial3Api::class)
            TopAppBar(
                title = {
                    Text(
                        text = "GurbaniLens",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.SemiBold,
                    )
                },
                actions = {
                    IconButton(onClick = onSettingsTap) {
                        Icon(
                            imageVector = Icons.Default.Settings,
                            contentDescription = "Settings",
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background,
                    titleContentColor = MaterialTheme.colorScheme.onBackground,
                ),
            )
        },
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .padding(horizontal = 24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center,
        ) {
            Text(
                text = "Bring the Bani into focus.",
                style = MaterialTheme.typography.headlineMedium,
                color = MaterialTheme.colorScheme.onBackground,
                textAlign = TextAlign.Center,
                modifier = Modifier.fillMaxWidth(),
            )
            Spacer(Modifier.height(24.dp))
            Text(
                text = "Tap and recite a Pangti. We'll find the Shabad.",
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
                modifier = Modifier.fillMaxWidth(),
            )
            Spacer(Modifier.height(64.dp))

            Button(
                onClick = onSearchTap,
                shape = CircleShape,
                colors = ButtonDefaults.buttonColors(
                    containerColor = MaterialTheme.colorScheme.primary,
                    contentColor = MaterialTheme.colorScheme.onPrimary,
                ),
                modifier = Modifier.size(160.dp),
            ) {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center,
                ) {
                    Icon(
                        imageVector = Icons.Default.Mic,
                        contentDescription = "Search by voice",
                        modifier = Modifier.size(72.dp),
                    )
                }
            }
            Spacer(Modifier.height(24.dp))
            Text(
                text = "Search by voice",
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onBackground,
                textAlign = TextAlign.Center,
                fontWeight = FontWeight.Medium,
            )
        }
    }
}

@Preview(showBackground = true, name = "Home — dark")
@Composable
private fun HomeScreenPreview() {
    GurbaniLensTheme(darkTheme = true) {
        HomeScreen(onSearchTap = {}, onSettingsTap = {})
    }
}

File: ios/GurbaniLens/GurbaniLens/UI/Home/HomeScreen.swift
import SwiftUI

/// v1 Home — single CTA: "Search by voice." Mirrors
/// `android/.../ui/home/HomeScreen.kt`.
struct HomeScreen: View {
    let onSearchTap: () -> Void
    let onSettingsTap: () -> Void

    var body: some View {
        NavigationStack {
            VStack(spacing: 24) {
                Spacer().frame(height: 8)
                Text("Bring the Bani into focus.")
                    .font(.system(size: 28, weight: .semibold))
                    .foregroundColor(Theme.onBackground)
                    .multilineTextAlignment(.center)
                Text("Tap and recite a Pangti. We'll find the Shabad.")
                    .font(.system(size: 17))
                    .foregroundColor(Theme.onSurfaceVariant)
                    .multilineTextAlignment(.center)
                Spacer().frame(height: 32)
                Button(action: onSearchTap) {
                    ZStack {
                        Circle().fill(Theme.primary)
                        Image(systemName: "mic.fill")
                            .font(.system(size: 64, weight: .medium))
                            .foregroundColor(Theme.onPrimary)
                    }
                    .frame(width: 160, height: 160)
                    .shadow(color: Theme.primary.opacity(0.4), radius: 24, y: 8)
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Search by voice")
                Text("Search by voice")
                    .font(.system(size: 18, weight: .medium))
                    .foregroundColor(Theme.onBackground)
                Spacer()
            }
            .padding(.horizontal, 24)
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .themed()
            .navigationTitle("GurbaniLens")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(action: onSettingsTap) {
                        Image(systemName: "gearshape").foregroundColor(Theme.onBackground)
                    }
                    .accessibilityLabel("Settings")
                }
            }
        }
    }
}
